pkg/provider/aws.go

```diff
@@ -60,3 +61,26 @@ func (p AWSProvider) GetParameterValue(name string, decrypt bool) (string, error
 
 	return *param.Parameter.Value, nil
 }
+
+func (p AWSProvider) GetParameterDataByPath(path string, decrypt bool) (map[string]string, error) {
+	// path is something like /path/to/env
+	params, err := p.Service.GetParametersByPath(&ssm.GetParametersByPathInput{
+		Path:           aws.String(path),
+		Recursive:      aws.Bool(true),
+		WithDecryption: aws.Bool(decrypt),
+	})
+
+	if err != nil {
+		log.Fatal("Failed to get params by path '%s': %s", path, err)
```

Comment:
I tested and it works!
However, it seems not to deal too well with permission errors.

```
time="2019-01-07T16:31:01Z" level=info msg="Using config: &amp;{eu-central-1 %!s(int=30)   0.0.0.0:9999 aws}"
time="2019-01-07T16:31:01Z" level=info msg=Running...
time="2019-01-07T16:31:01Z" level=info msg="Connected to cluster at https://xxxxxxx:443"
time="2019-01-07T16:31:01Z" level=info msg="Getting value for 'default/multi-secret'"
time="2019-01-07T16:31:01Z" level=fatal msg="Failed to get params by path '%s': %s/xxxxxxAccessDeniedException: User: arn:aws:iam::xxxxxxxxx:user/xxxxxxxxx is not authorized to perform: ssm:GetParametersByPath on resource: arn:aws:ssm:eu-central-1:xxxxxxx:parameter/xxxxxxxx\n\tstatus code: 400, request id: c9b32e66-de85-4580-a6f9-365b67275374"
```

After logging, correctly, the exception, the container dies. 

I am not familiar with go, but nothing else is logged after that...but it probably attempts to unpack "params".
I assume it should terminate here instead of continuing.

Reply:
I'm open to suggestions, but it seems most apps just crash out if the AWS permissions aren't configured. There's not much else to do if the AWS API call fails. I did update from `Fatal` to `Fatalf` to fix the formatting bug though - thanks!

Reply:
The problem is that in my case the credentials were not wrong, but they had a policy attached allowing to access SSM parameters starting with /somedirectory/ (trailing slash) while I had configured the secret annotation to "/somedirectory" (no trailing slash).
Granted, it was a misconfiguration on the secret, but seeing the aws-ssm pod crashlooping was a bit puzzling at first.  It can also be argued that to debug such a situation one would have to check the pod logs anyway, and the crashloop actually warned me that something was wrong. However, it prevented other secrets (for which no permission problem existed) to be updated, and since there is one single aws-ssm deployment on the cluster, serving potentially more than one application, the side effect of one broken secret in one namespace affecting operations on an unrelated application can be a bit problematic. 

I'm mostly thinking out loud here, I'm 100% grateful for your help, this is helping me solving me more than a problem :)

Reply:
> However, it prevented other secrets (for which no permission problem existed) to be updated, and since there is one single aws-ssm deployment on the cluster, serving potentially more than one application, the side effect of one broken secret in one namespace affecting operations on an unrelated application can be a bit problematic.

That's a good point - I'm going to open this as a separate issue to avoid too many changes at once. Thanks for your help!